src/domain_classes/dimension.py

```diff
@@ -3,53 +3,31 @@
 
 class Dimension:
     def __init__(self, dimensions: str, attribute_type: str):
+        """
+        dimensions: define the structure of the array. For example:
+        - "*" is a one dimensional array of any length.
+        - "*,*" is a two dimensional array of any length.
+        - "2,1" is a 2 dimensional array where inner arrays have length one - for example [[1], [3]]
+
+        type: define what type the values in the array have. Primitive types (bool, int, str, etc)
```

Comment:
```suggestion
        attribute_type: define what type the values in the array have. Primitive types (bool, int, str, etc)
```
Also, I miss an example of valid input. Should I pass in "int" or "integer"?

Reply:
ah, right. The primitive types should be written out in full: boolean, integer, string and number. WIll update the docstring.